packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -86,18 +86,31 @@ export class MetadataOutlineProvider
   public async getComponents(
     metadataType: BrowserNode
   ): Promise<BrowserNode[]> {
-    const username = this.defaultOrg!;
-    const cmpUtil = new ComponentUtils();
-    const componentsList = await cmpUtil.loadComponents(
-      username,
-      metadataType.label!
-    );
-    const nodeList = componentsList.map(
-      cmp => new BrowserNode(cmp, NodeType.MetadataCmp)
-    );
-    if (nodeList.length === 0) {
-      nodeList.push(
-        new BrowserNode(nls.localize('empty_components'), NodeType.EmptyNode)
+    let nodeList: BrowserNode[] = [];
+    if (TypeUtils.FOLDER_TYPES.has(metadataType.label!)) {
+      let typeName = metadataType.label!;
```

Comment:
I think this logic for getting the correct typeName for a metadataType that supports folders should in a util instead of having it here. Maybe having it in `TypesUtils`.